lib/services/database.go

```diff
@@ -341,6 +408,52 @@ func NewDatabaseFromMemoryDBCluster(cluster *memorydb.Cluster, extraLabels map[s
 		})
 }
 
+// MetadataFromAzureMySQLServer creates Azure metadata from the provided AzureMySQL server.
+func MetadataFromAzureMySQLServer(server *armmysql.Server) (*types.Azure, error) {
+	var (
+		name   string
+		region string
+		id     string
+	)
+	if server.Name != nil {
+		name = *server.Name
+	}
```

Comment:
What would you say about implementing a utility helper function like: 
```
// StringValue returns the value of the string pointer passed in or
// "" if the pointer is nil.
func StringValue(v *string) string {
	if v != nil {
		return *v
	}
	return ""
}
```

This type of logic seems to very common and the helper function will a bit improve readability. 
where code can be reduced to: 
```go
func MetadataFromAzureMySQLServer(server *armmysql.Server) (*types.Azure, error) {
  return &types.Azure{
		  Name:       StringValue(server.Name),
		  Region:     StringValue(server.Location),
		  ResourceID: StringValue(server.ID),
	  }, nil
}
```

Reply:
This code is obsolete since I refactored